### PR TITLE
ISPN-5662 Change stream timeout values for remote operations to use

### PR DESCRIPTION
**Pull request by wburns**
replication timeout

* Set timeout to replication timeout by default
* Allow overriding timeout value

https://issues.jboss.org/browse/ISPN-5662

### CODE REVIEW

File: core/src/main/java/org/infinispan/stream/impl/AbstractCacheStream.java [hunk 1]
```diff
@@ -262,8 +268,8 @@ public void close() {
          remoteResults.onCompletion(null, Collections.emptySet(), localValue);
          try {
             if ((earlyTerminatePredicate == null || !earlyTerminatePredicate.test(localValue)) &&
-                    !csm.awaitCompletion(id, 30, TimeUnit.SECONDS)) {
-               throw new CacheException(new TimeoutException());
+                    !csm.awaitCompletion(id, timeout, timeoutUnit)) {
+               new TimeoutException();
```
Comment (anistor) [inline core/src/main/java/org/infinispan/stream/impl/AbstractCacheStream.java]: Not thrown?

File: core/src/main/java/org/infinispan/stream/impl/AbstractCacheStream.java [hunk 2]
```diff
@@ -316,8 +322,8 @@ public void close() {
             }
             try {
                if ((!localRun || earlyTerminatePredicate == null || !earlyTerminatePredicate.test(localValue)) &&
-                       !csm.awaitCompletion(id, 30, TimeUnit.SECONDS)) {
-                  throw new CacheException(new TimeoutException());
+                       !csm.awaitCompletion(id,  timeout, timeoutUnit)) {
+                  new TimeoutException();
```
Comment (anistor) [inline core/src/main/java/org/infinispan/stream/impl/AbstractCacheStream.java]: Another one, and there seem to be more.

File: core/src/main/java/org/infinispan/stream/impl/AbstractCacheStream.java (hunk 1 above)
Comment (wburns) [inline core/src/main/java/org/infinispan/stream/impl/AbstractCacheStream.java]: Yeah find/replace failure :)

Comment (wburns) [inline core/src/main/java/org/infinispan/stream/impl/AbstractCacheStream.java]: Fixed, good catch.

**Comment by anistor** [comment]
Integrated in master. Thanks @wburns!

**Pull request closed without merging**